### PR TITLE
Issue #3155: Add 'info' type for backdrop_set_message

### PR DESCRIPTION
**Pull request by klonos**
https://github.com/backdrop/backdrop-issues/issues/3155

### CODE REVIEW

**Comment by klonos** [comment]
The test failure seems to be random ...closing and reopening to restart the tests.

File: core/includes/theme.inc [hunk 1]
```diff
@@ -1730,24 +1730,25 @@ function theme_status_messages($variables) {
...
+        $output .= "    <li>" . $message . "</li>\n";
       }
-      $output .= " </ul>\n";
+      $output .= "  </ul>";
     }
     else {
       $output .= reset($messages);
     }
-    $output .= "</div>\n";
+    $output .= "\n</div>\n";
```
Comment (opi) [inline core/includes/theme.inc:1751]: Nitpicking : `\n` are always printed out at the end of every line, so maybe restore this `\n` after the `</ul>` closing tag above.

Comment (klonos) [reply core/includes/theme.inc:1751]: This was intentional @opi ...you see, all these `\n`s and additional spaces that's going on here is in order to print the entire message `<div>` in a pretty format when one inspects the rendered code. With multiple messages of the same type, having the leading `\n` either way, renders things like so:

```
<div class="messages info">
  <h2 class="element-invisible">Info message</h2>
  <ul>
    <li>info1</li>
    <li>info2</li>
  </ul>
</div>
```

...so no problem either way. But when you have single messages, it prints them like so:

- with the `\n` after the `</ul>`:

```
<div class="messages info">
  <h2 class="element-invisible">Info message</h2>
single info message</div>
```

- with the `\n` before the `</div>`:

```
<div class="messages info">
  <h2 class="element-invisible">Info message</h2>
single info message
</div>
```

This is because moving the `\n` after the `</ul>` puts it inside the `if` statement, which is not true for single messages of the same type. This effectively makes it so that the `\n` is not printed in these cases, which makes the closing `</div>` tag print in the same line as the actual message text, which defeats the purpose of using all these `\n`s (which was to make the rendered code look pretty).

I hope that this makes sense @opi 

PS: In general, my preference would be to remove all this "witchcraft" from specific places like this, and have the entire code that is generated by Drupal/Backdrop be prettified on the fly, at a lower-level. But I guess that if possible, someone would have done it by now 😄

Comment (opi) [reply core/includes/theme.inc:1751]: wow, such an explanation ! And +1 for removing this tabs and line from the code. I never read raw sources anymore, browsers inspector always prettify the html now.

File: core/modules/system/css/messages.theme.css [hunk 2]
```diff
@@ -58,29 +58,54 @@
...
+.info a {
+  color: #004570;
+}
+div.info:before {
+  background-image: url(../../../misc/message-24-info.png);
+  background-color: #0074bd;
+}
+
+.messages a {
+  text-decoration: underline;
```
Comment (opi) [inline core/modules/system/css/messages.theme.css:108]: Just realized that links has no *hover* or *focus* effect. This is not nice regarding accessibility. 
The easiest could be to remove the text-decoration:
```
.messages a:hover,
.messages a:focus {
  text-decoration: none;
}
```

**Review by opi — CHANGES_REQUESTED** [review]

File: core/includes/theme.inc [hunk 3]
```diff
@@ -1730,24 +1730,25 @@ function theme_status_messages($variables) {
...
     if (!empty($status_heading[$type])) {
-      $output .= '<h2 class="element-invisible">' . $status_heading[$type] . "</h2>\n";
+      $output .= '  <h2 class="element-invisible">' . $status_heading[$type] . "</h2>\n";
     }
     if (count($messages) > 1) {
-      $output .= " <ul>\n";
+      $output .= "  <ul>\n";
       foreach ($messages as $message) {
-        $output .= '  <li>' . $message . "</li>\n";
+        $output .= "    <li>" . $message . "</li>\n";
```
Comment (jenlampton) [inline core/includes/theme.inc:1744]: Looks like there's an unintentional change here, that may result in a micro-performance regression, as `''` is a smidge faster than `""` in PHP.  It would probably be better to revert this change.

File: core/includes/theme.inc [hunk 4]
```diff
@@ -1730,24 +1730,25 @@ function theme_status_messages($variables) {
...
       }
-      $output .= " </ul>\n";
+      $output .= "  </ul>";
```
Comment (jenlampton) [inline core/includes/theme.inc:1746]: Looks like we lost the line-break here.

File: core/includes/theme.inc (hunk 1 above)
Comment (jenlampton) [reply core/includes/theme.inc:1751]: @klonos it sounds like we could add a line break to the single message then, so that it will match what happens with the multi-line message. I'd prefer not to rearrange the theme function too much, especially when that's not the point of this issue. I'm going to take a stab at a cleanup.

**Comment by jenlampton** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/2291
